### PR TITLE
engine_tirocks: port util

### LINKED ISSUE #13058: Use tirocks
## Development Task

[tirocks](https://github.com/busyjay/tirocks) is the new rocksdb wrapper, it provides more rusty and concise abstractions compared to existing tikv/rust-rocksdb.

Now the development of tirocks is about to finish, we can start migrate tikv to use tirocks.

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Close #13058 

What's Changed:

```commit-message
Some useless methods are not ported.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/sst_importer/src/import_mode.rs [hunk 1]
```diff
@@ -180,17 +180,17 @@ impl ImportModeDbOptions {
...
     soft_pending_compaction_bytes_limit: u64,
     hard_pending_compaction_bytes_limit: u64,
 }
 
 impl ImportModeCfOptions {
     fn optimized_for_import_mode(&self) -> Self {
         Self {
-            level0_stop_writes_trigger: self.level0_stop_writes_trigger.max(1 << 30),
-            level0_slowdown_writes_trigger: self.level0_slowdown_writes_trigger.max(1 << 30),
+            level0_stop_writes_trigger: -1,
```
Comment (BusyJay) [inline components/sst_importer/src/import_mode.rs:192]: /cc @overvenus  PTAL

File: components/engine_tirocks/src/cf_options.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    #[inline]
+    pub(crate) fn into_rocks(self) -> CfOptions {
+        match self {
+            RocksCfOptions::Rocks(opt) => opt,
+            _ => panic!("it's a titan cf option"),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn into_rocks_titan(self) -> TitanCfOptions {
```
Comment (tabokie) [inline components/engine_tirocks/src/cf_options.rs:40]:
```suggestion
    pub(crate) fn into_titan(self) -> TitanCfOptions {
```

File: components/engine_tirocks/src/cf_options.rs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    mem,
+    ops::{Deref, DerefMut},
+};
+
+use tirocks::{
+    option::{RawCfOptions, TitanCfOptions},
+    CfOptions,
+};
+
+pub enum RocksCfOptions {
```
Comment (tabokie) [inline components/engine_tirocks/src/cf_options.rs:13]: Maybe wrap it so that the enum type is private?

**Review by tabokie — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**